test/python/transpiler/test_consolidate_blocks.py

```diff
@@ -428,6 +429,91 @@ def test_identity_1q_unitary_is_removed(self):
         pm = PassManager([Collect2qBlocks(), Collect1qRuns(), ConsolidateBlocks()])
         self.assertEqual(QuantumCircuit(5), pm.run(qc))
 
+    def test_descent_into_control_flow(self):
+        """Test consolidation in blocks when control flow op is the same as at top level."""
+        qc = QuantumCircuit(2, 1)
+        u2gate1 = U2Gate(-1.2, np.pi)
+        u2gate2 = U2Gate(-3.4, np.pi)
+        qc.append(u2gate1, [0])
+        qc.append(u2gate2, [1])
+        qc.cx(0, 1)
+        qc.cx(1, 0)
+
+        pass_manager = PassManager()
+        pass_manager.append(Collect2qBlocks())
+        pass_manager.append(ConsolidateBlocks(force_consolidate=True))
+        result_top = pass_manager.run(qc)
+
+        qc_control_flow = QuantumCircuit(2, 1)
+        qc_block = copy.deepcopy(qc)
+
+        qc_block = QuantumCircuit(qc.qubits, qc.clbits)
+        qc_block.append(u2gate1, [0])
+        qc_block.append(u2gate2, [1])
+        qc_block.cx(0, 1)
+        qc_block.cx(1, 0)
+
+        ifop = IfElseOp((qc.clbits[0], False), qc_block, None)
+        qc_control_flow.append(ifop, qc.qubits, qc.clbits)
+
+        pass_manager = PassManager()
+        pass_manager.append(Collect2qBlocks())
+        pass_manager.append(ConsolidateBlocks(force_consolidate=True))
+        result_block = pass_manager.run(qc_control_flow)
+        gate_top = result_top[0].operation
+        gate_block = result_block[0].operation.blocks[0][0].operation
+        self.assertEqual(gate_top, gate_block)
+
+    def test_not_crossing_between_control_flow_block_and_parent(self):
+        """Test that consolidation does not occur across the boundary between control flow
+        blocks and the parent circuit."""
+        qc = QuantumCircuit(2, 1)
+        qc.cx(0, 1)
+        qc_true = QuantumCircuit(qc.qubits, qc.clbits)
+        qc_false = QuantumCircuit(qc.qubits, qc.clbits)
+        qc_true.cx(0, 1)
+        qc_false.cz(0, 1)
+        ifop = IfElseOp((qc.clbits[0], True), qc_true, qc_false)
+        qc.append(ifop, qc.qubits, qc.clbits)
+
+        pass_manager = PassManager()
+        pass_manager.append(Collect2qBlocks())
+        pass_manager.append(ConsolidateBlocks(force_consolidate=True))
+        qc_out = pass_manager.run(qc)
+
+        self.assertIsInstance(qc_out[0].operation, UnitaryGate)
+        self.assertTrue(np.alltrue(CXGate().to_matrix() == qc_out[0].operation.to_matrix()))
+        op_true = qc_out[1].operation.blocks[0][0].operation
+        op_false = qc_out[1].operation.blocks[1][0].operation
+        self.assertTrue(np.alltrue(CXGate().to_matrix() == op_true.to_matrix()))
+        self.assertTrue(np.alltrue(CZGate().to_matrix() == op_false.to_matrix()))
```

Comment:
It'd be cleaner here to use a Numpy matrix assertion (`np.testing.assert_allclose` or similar) so that on a test failure we get shown the matrices as the message; this form will just show us "False is not True".

Reply:
Done in 87b979e